radioactivedecay/utils.py

```diff
@@ -276,13 +286,67 @@ def build_nuclide_string(Z: int, A: int, meta_state: str = "") -> str:
     """
 
     if Z not in Z_DICT:
-        raise ValueError(str(Z) + " is not a valid atomic number")
+        raise ValueError(f"{Z} is not a valid atomic number")
 
     return_string = f"{Z_DICT[Z]}-{A}{meta_state}"
 
     return return_string
 
 
+class NuclideStrError(ValueError):
+    """
+    Custom exception class for invalid nuclide strings.
+
+    Parameters
+    ----------
+    nuclide : str
+        Nuclide string.
+    additional_message : str
+        Message with additional error context.
+
+    """
+
+    def __init__(self, nuclide: str, additional_message: str, *args) -> None:
+        super().__init__(args)
+        self.nuclide = nuclide
+        self.additional_message = additional_message
+
+    def __str__(self) -> str:
+        return (
+            f"{self.nuclide} is not a valid nuclide string. {self.additional_message}"
+        )
+
+
+def _process_metastable_element_str(metastable_element_str: str) -> Tuple[str, str]:
+    """
+    Function to separate a string that could either be an element string, or a combined metastable
+    state char + element string.
+
+    It separates the metastable state char correctly if the element is 2 chars (He, Li, Be, etc.).
+
+    If the element is 1 char (H, B, C, N, O, F, I, U etc.), ambiguities can occur, e.g. is 'ni'
+    Nickel or a second metastable state of Iodine. The atomic mass number & nuclear data is
+    needed to correctly separate these cases. However for simplicity it is assumed user has
+    capitalized correctly.
+
+    Note: bugs may occur with incorrect capitalizations. See test_utils.py for some failure cases.
+    """
+
+    # metastable_element_str is 3 chars: must contain metastable state + 2 char element string
+    if len(metastable_element_str) > 2:
+        return metastable_element_str[0], metastable_element_str[1:]
```

Comment:
Are we assuming the mass-first case will always have the metastable state before the element symbol? I.e. 99mTc but never 99Tcm?

Reply:
Yes we are. I don't recall ever seeing latter format used (i.e. 99Tcm), but it appears it's used sometimes: https://www.google.com/search?q="99Tcm"